### PR TITLE
iOS examples — extract core wrapper

### PR DESCRIPTION
**Pull request by StuartHarris**
This updates all 6 iOS examples to extract a "core" wrapper (the generic code that only changes when new capabilities are used) and to be more consistent with each other.

### CODE REVIEW

File: examples/bridge_echo/iOS/BridgePerf/ContentView.swift [hunk 1]
```diff
@@ -1,53 +1,27 @@
...
-    case event(Event)
-}
-
-@MainActor
-class Model: ObservableObject {
-    @Published var view = ViewModel(count: 0)
-
-    init() {
+struct ContentView: View {
+    @ObservedObject var model: Core
```
Comment (charypar-nm) [inline examples/bridge_echo/iOS/BridgePerf/ContentView.swift:5]: maybe `var core`?

File: examples/bridge_echo/iOS/BridgePerf/core.swift [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+@MainActor
+class Core: ObservableObject {
+    @Published var view: ViewModel
+    
+    init() {
+        self.view = try! .bincodeDeserialize(input: [UInt8](BridgePerf.view()))
+    }
+    
+    func update(event: Event) {
```
Comment (charypar-nm) [inline examples/bridge_echo/iOS/BridgePerf/core.swift:12]: Do we want to be consistent on whether we pass single arguments as `_` or not?

File: examples/bridge_echo/iOS/BridgePerf/ContentView.swift (hunk 1 above)
Comment (charypar) [inline examples/bridge_echo/iOS/BridgePerf/ContentView.swift:5]: Maybe `var core`?

File: examples/bridge_echo/iOS/BridgePerf/core.swift (hunk 2 above)
Comment (charypar) [inline examples/bridge_echo/iOS/BridgePerf/core.swift:12]: Do we want to be consistent in whether we pass single aguments named or as `_`?

File: examples/notes/iOS/Notes/ContentView.swift [hunk 3]
```diff
@@ -19,70 +12,33 @@ extension TextCursor {
...
-    init() {
-        view = ViewModel(text: "", cursor: .position(0)) // this will get replaced by the next call
+struct ContentView: View {
+    @ObservedObject var model: Core
 
+    init(model: Core) {
+        self.model = model
         // Insert initial document for testing
-        update(event: .event(.insert("Hello from the core!")))
+        model.update(event: .insert("Hello!"))
```
Comment (charypar) [inline examples/notes/iOS/Notes/ContentView.swift:21]: Is it worth putting update into environment, like in the tap to pay example?

**Review by charypar — APPROVED** [review]
Looks good. A few questions, but nothing major. 

Decide what you wanna do about those, but ship it either way :shipit:

File: examples/notes/iOS/Notes/ContentView.swift (hunk 3 above)
Comment (StuartHarris) [reply examples/notes/iOS/Notes/ContentView.swift]: I think it's deffo worth doing when we have child components, but otherwise probs not

**Pull request merged**